types/src/main.ts

```diff
@@ -1656,7 +1696,7 @@ export interface SignatureHelp {
  * For most programming languages there is only one location at which a symbol is
  * defined. If no definition can be found `null` is returned.
  */
-export type Definition = Location | Location[] | null;
+export type Definition = Location | Location[] | DefinitionLink | DefinitionLink[] | null;
```

Comment:
The deeper question that we need to decide is whether we want to deprecate Location for definitions and ask for providing a DefinitionLink instead. If so we shouldn't or them into the Defintion, but rather list them in the results.

Location is semantically equal to DefinitionLink if originSelectionRange and targetSelectionRange are not provided. 

What is your opinion?

Reply:
@dbaeumer Your suggestion seems reasonable to me. We can add a deprecation notice to `Definition` then to try to help discourage people from using it.

It is too bad that we cannot simply extend `DefinitionLink` from `Location` as the attribute names don't match (and `range` in `Location` as-is is a little ambiguous compared to the more specific names in `DefinitionLink`).

Reply:
It's not clear why, but in the VS Code API, @jrieken thought it would not be a good idea to "mix `DefinitionLink` into `Definition`" as is being done here. See Microsoft/vscode#54424

Reply:
@rcjsuen then lets keep them different. I talked to @jrieken about why he kept them different but there was no deeper semantic meaning behind it.